### PR TITLE
fetch users using UseEffect in AssignmentList

### LINKED ISSUE #699: AssignmentList calls `fetchUser` too many times
As `fetchUser` is not currently encapsulated using `useEffect` in AssignmentList, it often issues far too many requests to the backend on loading the source page. 

image

### PR DESCRIPTION
**Pull request by dannygoldstein**
Fixes #699.

### CODE REVIEW

**Comment by dannygoldstein** [comment]
effectively Green @acrellin  (passed GA, but not travis).

File: static/js/components/AssignmentList.jsx [hunk 1]
```diff
@@ -16,6 +16,22 @@ const AssignmentList = ({ assignments }) => {
...
+  unique_requester_ids.sort((a, b) => a - b);
+
+  // use useEffect to only send 1 fetchUser per User
+  useEffect(() => {
+    unique_requester_ids.forEach((id) => {
+      if (!users[id]) {
+        dispatch(UserActions.fetchUser(id));
+      }
+    });
+  }, [...unique_requester_ids, dispatch]);
```
Comment (acrellin) [inline static/js/components/AssignmentList.jsx:33]: Any reason you're spreading unique requester ids in the dependencies array? You just need to specify the names of the variables that the hook uses. Also, looks like `users` should be added here.

File: static/js/components/AssignmentList.jsx [hunk 2]
```diff
@@ -16,6 +16,22 @@ const AssignmentList = ({ assignments }) => {
   const { observingRunList } = useSelector((state) => state.observingRuns);
   const { instrumentList } = useSelector((state) => state.instruments);
 
+  // fetch all the requester ids before rendering the component
+  const requester_ids = assignments.map(
```
Comment (acrellin) [inline static/js/components/AssignmentList.jsx:20]:
```suggestion
  const requesterIds = assignments.map(
```

File: static/js/components/AssignmentList.jsx [hunk 3]
```diff
@@ -16,6 +16,22 @@ const AssignmentList = ({ assignments }) => {
...
+    (assignment) => assignment.requester_id
+  );
+  const unique_requester_ids = [...new Set(requester_ids)];
```
Comment (acrellin) [inline static/js/components/AssignmentList.jsx:23]:
```suggestion
  const uniqueRequesterIds = [...new Set(requester_ids)];
```

**Pull request merged**